src/vs/workbench/contrib/terminal/browser/links/terminalLinkDetectorAdapter.ts

```diff
@@ -0,0 +1,110 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Emitter } from 'vs/base/common/event';
+import { Disposable } from 'vs/base/common/lifecycle';
+import { localize } from 'vs/nls';
+import { IInstantiationService } from 'vs/platform/instantiation/common/instantiation';
+import { ITerminalSimpleLink, ITerminalLinkDetector, TerminalLinkType } from 'vs/workbench/contrib/terminal/browser/links/links';
+import { TerminalLink } from 'vs/workbench/contrib/terminal/browser/links/terminalLink';
+import { XtermLinkMatcherHandler } from 'vs/workbench/contrib/terminal/browser/links/terminalLinkManager';
+import { IBufferLine, ILink, ILinkProvider, IViewportRange } from 'xterm';
+
+export interface IShowHoverEvent {
+	link: TerminalLink;
+	viewportRange: IViewportRange;
+	modifierDownCallback?: () => void;
+	modifierUpCallback?: () => void;
+}
+
+/**
+ * Wrap a link detector object so it can be used in xterm.js
+ */
+export class TerminalLinkDetectorAdapter extends Disposable implements ILinkProvider {
```

Comment:
crying at the wordiness of `DetectorAdapter` 😭  not sure if there's a better one though

Reply:
Yeah I'm happy enough with it, it's best practice for design patterns to include the names they use, that way it's super clear what this file/class does just from the name